src/components/bookmark-list.ts

```diff
@@ -46,15 +57,31 @@ export class BookmarkList extends LitElement {
   // State controller for persistence with automatic observation
   private stateController = new StateController<BookmarkListState>(this, {
     storageKey: 'bookmark-list-state',
-    defaultState: { selectedFilter: 'all', scrollPosition: 0 },
-    observedProperties: ['selectedFilter', 'scrollPosition'],
+    defaultState: { 
+      selectedFilter: 'all', 
+      scrollPosition: 0,
+      pagination: {
+        all: { currentPage: 1 },
+        unread: { currentPage: 1 },
+        archived: { currentPage: 1 }
+      }
```

Comment:
Is this used for anything or can it be removed? I don't think we need to keep track of pagination separately for each filter. We only need to know the pagination state for the currently-selected filter. When selecting a different filter, it's ok if the pagination restarts from zero.